CRM/Mailing/BAO/Component.php

```diff
@@ -50,7 +50,7 @@ public function __construct() {
    * @return CRM_Core_BAO_LocationType.
    */
   public static function retrieve(&$params, &$defaults) {
-    $component = new CRM_Mailing_DAO_Component();
+    $component = new CRM_Mailing_BAO_Component();
```

Comment:
shouldn't we put the new DAO here - given the BAO needs renaming too but perhaps in a later iteration?

Reply:
I agree. @pradpnayak can you update?